scripts/build/builders/nxp/builder.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+from ..gn import GnBuilder
+
+
+def check_modifiers(modifiers, supported_modifiers):
```

Comment:
Overall I find this loose dictionary approach adds too much complexity.

At a minimum please add type annotations to things. I would also say that if you have `Dict[str,str]` for example or other things like that, it is probably too generic and complex.